File: Easy/463_IslandPerimeter/Python3/Solution.py
class Solution:
    def islandPerimeter(self, grid: List[List[int]]) -> int:
        perimiter = 0
        for row in range(0, len(grid)):
            for col in range(0, len(grid[0])):
                if grid[row][col] == 1:
                    if row - 1 < 0 or grid[row - 1][col] == 0: perimiter += 1
                    if row + 1 >= len(grid) or grid[row + 1][col] == 0: perimiter += 1
                    if col - 1 < 0 or grid[row][col - 1] == 0: perimiter += 1
                    if col + 1 >= len(grid[0]) or grid[row][col + 1] == 0: perimiter += 1
        return perimiter

File: Easy/463_IslandPerimeter/C++/Solution.cpp
#include <vector>

class Solution {
public:
    int islandPerimeter(std::vector<std::vector<int>>& grid) {
        int perimeter = 0;

        for (int i = 0; i < grid.size(); i++) {
            for (int j = 0; j < grid[0].size(); j++) {
                if (grid[i][j]) {
                    if (i == 0 || !grid[i - 1][j]) perimeter++;
                    if (i == grid.size() - 1 || !grid[i + 1][j]) perimeter++;
                    if (j == 0 || !grid[i][j - 1]) perimeter++;
                    if (j == grid[0].size() - 1 || !grid[i][j + 1]) perimeter++;
                }
            }
        }

        return perimeter;
    }
};